検索フォームの全てを実装（「種目選択」「都道府県選択」「対象年齢」の検索オプションを追加）

toshinori-m opened the pull request:
### 変更内容
競技検索機能の拡張として、「種目選択」「都道府県選択」「対象年齢」の選択機能を追加しました。

- 「種目選択」: 選択した競技に応じて表示されるように追加
- 「都道府県選択」: ユーザーが地域を指定して検索できるように追加
- 「対象年齢」: ユーザーが特定の年齢層向けの競技を検索できるように追加

close https://github.com/toshinori-m/stay_connect/issues/176

toshinori-m committed:
feat: 「種目選択」「都道府県選択」「対象年齢」の検索オプションを追加

hirosi1900day commented on frontend-react/src/pages/HomePage.tsx:
```diff
@@ -7,30 +7,72 @@ import { SelectOption } from "@/types"
+  
+    fetchData()
+  }, [apiClient])
+
```
fetchDataをuseEffectの中で定義しているのは何か意図がありますか？？
個人的には外で良い気がしているのですが

またapiClientの状態はほぼ変わらない気がするのでapiClientが依存関係に含まれてるのはどこのタイミングでこのuseEffectが実行されるのか直感的に分かりにくい気がします(初期レンダリング時に一覧を取得したいのであれば[]を依存にする方が分かりやすい)

toshinori-m replied:
初期レンダリング時に一覧を取得したいので、依存配列を [] に修正しました。その場合に下の警告が表示されますが、これは無視しても大丈夫でしょうか？

`React Hook useEffect has a missing dependency: 'fetchData'. Either include it or remove the dependency array.`
もし適切な解決策があれば、教えていただけると助かります！

hirosi1900day commented on frontend-react/src/pages/HomePage.tsx:
```diff
@@ -7,30 +7,72 @@ import { SelectOption } from "@/types"
+  }, [apiClient])
+
   useEffect(() => {
     const fetchSportsTypes = async () => {
```
こちらもuseEffectの中でfetchSportsTypes定義するのは何か意図がありますか？？
sportsTypeSelectedが変更されるたびに関数が生成されてしまう気がします

hirosi1900day commented on frontend-react/src/pages/HomePage.tsx:
```diff
@@ -7,30 +7,72 @@ import { SelectOption } from "@/types"
   useEffect(() => {
     const fetchSportsTypes = async () => {
       try {
         setErrors([])
```
ここでsetErrorsをリセットすると例えばfetchDataとfetchSportsTypesで両方エラーになった時にどちらかのエラーメッセージがリセットされません？？

hirosi1900day commented on frontend-react/src/pages/HomePage.tsx:
```diff
@@ -7,30 +7,72 @@ import { SelectOption } from "@/types"
 
     fetchSportsTypes()
-  }, [apiClient])
+  }, [sportsTypeSelected, apiClient])
```
apiClientは依存関係に入れる必要はありますか？？
sportsTypeSelectedが変更されたらapiにリクエストしたいが目的だと思うので、無くした方が直感的に分かりやすいですし、不要なapiリクエストが減らせるかなと思いましたがいかがでしょうか？

toshinori-m replied:
依存配列を [sportsTypeSelected] に修正しました。その場合に下の警告が表示されますが、これは無視しても大丈夫でしょうか？

`React Hook useEffect has a missing dependency: 'fetchSportsTypes'. Either include it or remove the dependency array.`
もし適切な解決策があれば、教えていただけると助かります！

hirosi1900day replied:
不要なlintの指摘はignoreしてください
https://zenn.dev/mackay/articles/1e8fcce329336d

toshinori-m replied:
コード全体でESLintのルールを無効にする方法で警告表示を下の方法で無くそうとしましたが、出来なかったのでコメントでignoreする方法で警告コメントを無くしています！

- docker compose exec frontend-react yarn eslint . → エラーなし
- "ESLint: Show Output Channel" →何も表示されない
- settings.json に eslint.nodePath を追加し、VSCode を再起動→変化なし
- eslint-plugin-react-hooks@5.2.0 をインストール→変化なし
- .eslintrc.jsを下のようにしルートディレクトリに追加→変化なし
```
module.exports = {
  root: true,
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: "module"
  },
  extends: [
    "react-app",
    "react-app/jest",
    "plugin:react-hooks/recommended"
  ],
  rules: {
    "react-hooks/exhaustive-deps": "off"
  }
};
```

hirosi1900day replied:
ファイル単位でignoreすると本来不適切な箇所も指摘されなくなるためコメントでignoreすることを基本としてください

toshinori-m committed:
Refactor: useEffect の関数定義を外部に移動し、依存関係を最適化

- apiClient を useEffect の依存配列から削除し、不要な API リクエストを削減
- エラーメッセージのリセット処理を見直し、複数のエラーが正しく表示されるよう修正

hirosi1900day commented on frontend-react/src/pages/HomePage.tsx:
```diff
@@ -7,30 +7,74 @@ import { SelectOption } from "@/types"
+    } catch (error: unknown) {
+      console.error("APIエラー発生:", error)
+      setErrors(prev => [...prev, "スポーツ種目の取得に失敗しました"])
+      apiErrorHandler(error, setErrors)
```
apiErrorHandleの中でsetErrorsの配列を上書きする動きに見えるのでまだエラーが消えてしまいませんかね？
apiErrorHandlerの引数にsetErrorsを指定すると汎用メソッドとしては取り回しが悪いのでapiErrorHandlerの戻り値はエラーの配列を返す形にした方がシンプルで且つ負債化しにくかなとおみましたがいかがでしょうか？？

toshinori-m committed:
refactor: apiErrorHandlerの戻り値をエラー配列に変更し、setErrorsを直接渡さないよう修正
